omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskActionResource2_4.java

```diff
@@ -0,0 +1,217 @@
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Creatable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.scheduler.TaskDefinition;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.BooleanProperty;
+import io.swagger.models.properties.StringProperty;
+
+@Resource(name = RestConstants.VERSION_1
+		+ "/taskaction", supportedClass = TaskAction.class, supportedOpenmrsVersions = { "2.4.*" })
+public class TaskActionResource2_4 extends BaseDelegatingResource<TaskAction> implements Creatable {
+
+	private TaskServiceWrapper2_4 taskServiceWrapper = new TaskServiceWrapper2_4();
+
+	public void setTaskServiceWrapper(TaskServiceWrapper2_4 taskServiceWrapper) {
+		this.taskServiceWrapper = taskServiceWrapper;
+	}
+
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		TaskAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+
+		Collection<TaskDefinition> taskDefinitions;
+		if (action.isAllTasks()) {
+			taskDefinitions = taskServiceWrapper.getRegisteredTasks();
+			action.setTasks(new ArrayList<TaskDefinition>(taskDefinitions));
+		} else {
+			taskDefinitions = action.getTasks();
+		}
+
+		TaskAction.Action actionType = action.getAction();
+		if (actionType != TaskAction.Action.RESCHEDULEALLTASKS) {
+			if (taskDefinitions == null || taskDefinitions.isEmpty()) {
+				throw new IllegalRequestException(
+						"Cannot execute action " + actionType + " on empty set of task definitions.");
+			}
+		}
+
+		switch (action.getAction()) {
+		case SCHEDULETASK:
+			scheduleTasks(taskDefinitions);
+			break;
+		case RESCHEDULETASK:
+			reScheduleTasks(taskDefinitions);
+			break;
+		case DELETE:
+			deleteTasks(taskDefinitions);
+			break;
+		case SHUTDOWNTASK:
+			shutDownTasks(taskDefinitions);
+			break;
+		case RESCHEDULEALLTASKS:
+			reScheduleAllTasks();
+			break;
+		case RUNTASK:
+			runTasks(taskDefinitions);
+			break;
```

Comment:
These `case` statements should be indented one more level.

Reply:
yes sure

Reply:
yes sure,going to add it